src/shogun/base/SGObject.h

```diff
@@ -1306,13 +1315,30 @@ void CSGObject::observe(
 	this->observe(obs);
 }
 
-template <class T>
+template<class T,
+		typename std::enable_if_t<
+				!is_sg_vector<T>::value &&
+				!is_sg_matrix<T>::value>*>
 void CSGObject::observe(const int64_t step, const std::string& name) const
 {
 	auto param = this->get_parameter(BaseTag(name));
 	auto obs = some<ObservedValueTemplated<T>>(
 		step, name, any_cast<T>(param.get_value()), param.get_properties());
 	this->observe(obs);
 }
+
+template<class T,
+		typename std::enable_if_t<
+				is_sg_vector<T>::value ||
+				is_sg_matrix<T>::value>*>
+void CSGObject::observe(const int64_t step, const std::string& name) const
+{
+	auto param = this->get_parameter(BaseTag(name));
+	auto obs = some<ObservedValueTemplated<T>>(
```

Comment:
I also wonder: couldn't we do this inside the `ObservedValueTemplated` class?

Reply:
I would prefer to keep a separate method to construct an `ObservedValueTemplated` object from a `Tag`.

Reply:
If you are talking about a method to construct a new `ObservedValueTemplated` :)